code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/EnsureSettingForProvider.java

```diff
@@ -0,0 +1,73 @@
+import io.cattle.platform.util.type.InitializationTask;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class EnsureSettingForProvider implements InitializationTask {
+
+    private static final Log logger = LogFactory.getLog(EnsureSettingForProvider.class);
+
+    @Inject
+    SettingsUtils settingsUtils;
+
+    @Inject
+    GithubUtils githubUtils;
+
+    @Override
+    public void start(){
+        if (SecurityConstants.SECURITY.get() &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_SECRET.get()) &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_ID.get())){
+            if (StringUtils.isBlank(SecurityConstants.AUTH_PROVIDER.get())) {
+                settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, GithubConstants.CONFIG);
+                logger.info("Upgrade detected. Adding setting for :" + SecurityConstants
+                        .AUTH_PROVIDER_SETTING);
+
+            }
+            if ("restricted".equalsIgnoreCase(GithubConstants.ACCESS_MODE.get()) &&
+                    StringUtils.isBlank(GithubConstants.ALLOWED_IDENTITIES_SETTING)){
+                List<Identity> allowedIdentities = new ArrayList<>();
+                String allowedUsers = ArchaiusUtil.getString("api.auth.github.allowed.users").get();
+                List<String> userIds = githubUtils.fromCommaSeparatedString(allowedUsers);
+                for (String userId: userIds){
+                    Identity identity = new Identity(GithubConstants.USER_SCOPE, userId.split(":")[1]);
+                    allowedIdentities.add(identity);
+                }
+                String allowedOrgs = ArchaiusUtil.getString("api.auth.github.allowed.orgs").get();
+                List<String> orgIds = githubUtils.fromCommaSeparatedString(allowedOrgs);
+                for (String orgId: orgIds){
+                    Identity identity = new Identity(GithubConstants.ORG_SCOPE, orgId.split(":")[1]);
+                    allowedIdentities.add(identity);
+                }
+                Iterator<Identity> identityIterator = allowedIdentities.iterator();
+                StringBuilder sb= new StringBuilder();
+                while (identityIterator.hasNext()){
+                    sb.append(identityIterator.next().getId().trim());
+                    if (identityIterator.hasNext()) sb.append(',');
+                }
+                settingsUtils.changeSetting(GithubConstants.ALLOWED_IDENTITIES_SETTING, sb.toString());
+                logger.info("Upgrade detected. Adding setting for :" + GithubConstants.ALLOWED_IDENTITIES_SETTING);
+            }
+        } else {
+            settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, SecurityConstants.NO_PROVIDER);
```

Comment:
I think this else would get hit if I enabled LDAP and then restarted rancher/server because its correspond if statement would evaluate to false:

```
if (SecurityConstants.SECURITY.get() && # True
  StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_SECRET.get()) && # False
  StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_ID.get())){  # False
```